schema.json

```diff
@@ -0,0 +1,496 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "type": "object",
+    "required": [
+        "format_version",
+        "generated_at_timestamp",
+        "steplib_source",
+        "download_locations",
+        "assets_download_base_uri",
+        "steps"
+    ],
+    "additionalProperties": false,
+    "properties": {
+        "format_version": {
+            "type": "string",
+            "pattern": "^\\d+.\\d+.\\d+$",
+            "examples": [
+                "1.0.0"
+            ]
+        },
+        "generated_at_timestamp": {
+            "type": "integer",
+            "examples": [
+                1584720375
+            ]
+        },
+        "steplib_source": {
+            "type": "string",
+            "pattern": "^https://github.com/bitrise-io/bitrise-steplib.git$",
+            "examples": [
+                "https://github.com/bitrise-io/bitrise-steplib.git"
+            ]
+        },
+        "download_locations": {
+            "type": "array",
+            "minItems": 2,
+            "maxItems": 2,
+            "uniqueItems": true,
+            "items": {
+                "type": "object",
+                "examples": [
+                    {
+                        "src": "https://bitrise-steplib-collection.s3.amazonaws.com/step-archives/",
+                        "type": "zip"
+                    },
+                    {
+                        "src": "source/git",
+                        "type": "git"
+                    }
+                ],
+                "required": [
+                    "type",
+                    "src"
+                ],
+                "additionalProperties": false,
+                "properties": {
+                    "src": {
+                        "type": "string",
+                        "pattern": "(^https://bitrise-steplib-collection.s3.amazonaws.com/step-archives/$)|(^source/git$)"
+                    },
+                    "type": {
+                        "type": "string",
+                        "pattern": "(^git$)|(^zip$)"
+                    }
+                }
+            }
+        },
+        "assets_download_base_uri": {
+            "type": "string",
+            "pattern": "^https://bitrise-steplib-collection.s3.amazonaws.com/steps$",
+            "examples": [
+                "https://bitrise-steplib-collection.s3.amazonaws.com/steps"
+            ]
+        },
+        "steps": {
```

Comment:
please add a check for step IDs as they are required also.
- it is not empty
- contains specific set of characters